contracts/FreakTeam.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+
+
+enum Role{it, supportIt, intern}
+enum Skill{novice, advanced, competent, profiency, expert}
+
+struct Freak {
+        string name;
+        uint256 startDate;
+        uint256 employeeNumber;
+        Role choiceRole;
+        Skill choiceSkill;
+
+}
+
+
+contract FreakTeam is ERC1155, AccessControl {
+
+    mapping(address => Freak)  public freaks;
+    address[] public freakAccts;
+
+    bytes32 public constant FINANCIAL_ROLE = keccak256("FINANCIAL");
+    bytes32 public constant HR_ROLE = keccak256("HR_ROLE");
+    bytes32 public constant FREAK_ROLE = keccak256("FREAK_ROLE");
+
+    event addedFreak(address _address, string _name, uint256 startDate, uint256 employeeNumber, Role role, Skill skill);
+
+    constructor (address hr, address financial) ERC1155(""){
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(HR_ROLE, hr);
+        _setupRole(FINANCIAL_ROLE, financial);
+        _setRoleAdmin(FREAK_ROLE, HR_ROLE);
+    }
+
+
+    function addNewFreak(address _address, string memory _name,uint256 _startDate, uint256 _employeeNumber, Role _role, Skill _skill) external {
+        require(hasRole(HR_ROLE, msg.sender), "Caller is not a hr");
+        grantRole(FREAK_ROLE, _address);
+        freaks[_address].name = _name;
```

Comment:
There must be an easier way to add a new struct to a mapping instead of accessing the mapping entry for each property. What if the struct has 100 properties? 
Each access in the mapping cost gas which costs the user a bit more ETH.

Reply:
better?
        freaks[_address] = Freak(
            {
                name: _name,
                startDate: _startDate,
                stopDate: 0,
                employeeNumber: _employeeNumber,
                role: _role,
                skill: _skill
            }
        );
        freakAccounts.push(_address);

Reply:
Even shorter:
```        freaks[_address] = Freak(
            _name,
            _startDate,
            _employeeNumber,
            _role,
            _skill
        );
        ```

Reply:
I need to add the value 0 (for the endDate)       
 freaks[_address] = Freak(
                _name,
                _startDate,
                0,
                _employeeNumber,
                _role,
                _skill
        );